### PR TITLE
Update to libraries-bom 26.20.0

### PR DESCRIPTION
**Pull request by bvolpato**
Update `google_cloud_platform_libraries_bom` to `26.20.0`. 

Associated updates to `gax_version`, `google_cloud_spanner_version`, `google_code_gson_version`, and `grpc_version` following versions from https://mvnrepository.com/artifact/com.google.cloud/libraries-bom/26.20.0 (being indexed).

Attaching the effective pom generated with:
```
wget https://repo1.maven.org/maven2/com/google/cloud/libraries-bom/26.20.0/libraries-bom-26.20.0.pom

mvn help:effective-pom -f libraries-bom-26.20.0.pom
```

[libraries-bom-26.20.0-effective-pom.txt](https://github.com/apache/beam/files/12197833/libraries-bom-26.20.0-effective-pom.txt)

### CODE REVIEW

**Comment by bvolpato** [comment]
Run Java PreCommit

**Comment by bvolpato** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by bvolpato** [comment]
Run Java_Pulsar_IO_Direct PreCommit

**Comment by bvolpato** [comment]
Run SQL PreCommit

**Comment by bvolpato** [comment]
Run SQL_Java17 PreCommit

**Comment by bvolpato** [comment]
Run SQL_Java11 PreCommit

**Comment by bvolpato** [comment]
Run Java PreCommit

**Comment by bvolpato** [comment]
Run SQL PreCommit

**Comment by bvolpato** [comment]
Run SQL_Java11 PreCommit

**Comment by bvolpato** [comment]
Run SQL_Java17 PreCommit

**Review by Abacn — APPROVED** [review]

**Comment by bvolpato** [comment]
Investigating a few failures

**Comment by Abacn** [comment]
I see, `Could not resolve com.google.guava:guava:32.1.1-jre.`

**Comment by bvolpato** [comment]
Bumped version to be the same and giving it a try.

There's some weird handling of Guava that I am not familiar with. Both in terms of `com.google.guava:listenablefuture` and some handling to ignore any Android versions https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L1899-L1904

**Comment by bvolpato** [comment]
`com.google.guava:guava:32.1.1-jre` is valid, but it seems that it's being rejected by CI. Will need to take a more careful look

**Comment by bvolpato** [comment]
Found the reason: https://github.com/google/guava/releases/tag/v32.1.0#user-content-overlap

Going to try the suggested fix

**Comment by bvolpato** [comment]
Run Java_PVR_Flink_Docker PreCommit

**Comment by bvolpato** [comment]
Run GoPortable PreCommit

**Comment by bvolpato** [comment]
Run Python_PVR_Flink PreCommit

**Comment by bvolpato** [comment]
@mutianf @suztomo Are you familiar with the changes that are happening on Bigtable?

Seems like it's hitting a real regression

```
16:15:38 > Task :sdks:java:extensions:sql:test
16:15:38 
16:15:38 org.apache.beam.sdk.extensions.sql.meta.provider.bigtable.BigtableTableFlatTest > testSegementedInsert FAILED
16:15:38     java.lang.AssertionError at BigtableTableFlatTest.java:164
16:15:38         Caused by: java.lang.AssertionError at MatcherAssert.java:18
16:15:38 
16:15:58 715 tests completed, 1 failed, 8 skipped

```


`readTransformWithSegment` (proxy to `.withMaxBufferElementCount(2)`) is apparently dropping records.

**Comment by mutianf** [comment]
@bvolpato The test failure is caused from the emulator version bump. I'll look into it on Monday. I think we can revert the emulator version bump so the test can pass.

**Comment by bvolpato** [comment]
> @bvolpato The test failure is caused from the emulator version bump. I'll look into it on Monday. I think we can revert the emulator version bump so the test can pass.

@mutianf Thanks, doing it! The bump was to match libraries-bom, but perhaps this is the reason why a very old version was pinned.

**Comment by bvolpato** [comment]
Run Spotless PreCommit

**Comment by bvolpato** [comment]
Run CommunityMetrics PreCommit

**Comment by bvolpato** [comment]
Run Spotless PreCommit

**Comment by bvolpato** [comment]
Oof, this was more complicated than I expected.

Guava introduced some complications https://github.com/google/guava/releases#user-content-duplicate-ListenableFuture

It went as far as hanging/infinite looping Gradle in the classpath resolution when importing Truth, which is built against `32.0.1-android`. Excluding it for now, and can bump + remove TODOs when Truth 1.1.6 is out (bump was done already https://github.com/google/truth/commit/3f00bfefb6ea8de8447c2cffeea938588039986e).

Also, `google-cloud-bigtable-emulator` could not be updated to `0.162.1` (to match `libraries-bom`) due to a regression in a testcase. Bigtable team is taking a look at this.


@Abacn Can you have a new look since this changed quite a bit since the last time you reviewed? Thanks!

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -658,7 +658,7 @@ class BeamModulePlugin implements Plugin<Project> {
         google_api_services_bigquery                : "com.google.apis:google-api-services-bigquery:v2-rev20220924-$google_clients_version",
         google_api_services_cloudresourcemanager    : "com.google.apis:google-api-services-cloudresourcemanager:v1-rev20220828-$google_clients_version",
         google_api_services_dataflow                : "com.google.apis:google-api-services-dataflow:v1b3-rev20220920-$google_clients_version",
-        google_api_services_healthcare              : "com.google.apis:google-api-services-healthcare:v1-rev20230510-$google_clients_version",
```
Comment (Abacn) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:661]: rebase issue? #27720 got reverted here

**Review by Abacn — COMMENTED** [review]
Thanks for the quick fix!

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 1 above)
Comment (bvolpato) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: My bad -- reverted. Thanks for noticing

**Review by Abacn — APPROVED** [review]

**Pull request merged**